src/javascript/gameInstance.js

```diff
@@ -101,6 +102,32 @@ export default class GameInstance {
     }
   }
 
+  setUserKeyBind(trackKeyBind) {
+    const keyMap = this.vm.keyMap;
+    Logger.log("!!!", keyMap);
+    if (keyMap) {
+      this.userKeyBind = trackKeyBind.map((e) => keyMap[e]);
+      let ret = {};
+      Object.keys(keyMap).forEach((key) => {
+        ret[keyMap[key]] = key;
+      });
+      this.reverseKeyMap = ret;
+      Logger.log("keyMap:", this.userKeyBind, this.reverseKeyMap);
+    } else {
+      this.userKeyBind = trackKeyBind;
+      this.reverseKeyMap = null;
+    }
+  }
+
+  getKeyName(keyEvent) {
+    const key = keyEvent.key.toLowerCase();
+    if (this.reverseKeyMap) {
+      return this.reverseKeyMap[key] ?? "";
+    } else {
+      return key;
+    }
```

Comment:
If you use the empty map approach as described previously, this is simply

```suggestion
    return this.reverseKeyMap[key] ?? key;
```

with the added benefit that if the `reverseKeyMap[key]` lookup gives you a miss, it falls back to the user supplied key instead?